tools/link.py

```diff
@@ -1415,6 +1417,11 @@ def phase_linker_setup(options, state):  # noqa: C901, PLR0912, PLR0915
       # The following symbols need exposing to load and bootstrap the audio worklet:
       settings.INCOMING_MODULE_JS_API += ['instantiateWasm', 'wasm', 'wasmMemory']
 
+  if 'preRun' in settings.INCOMING_MODULE_JS_API and not settings.MINIMAL_RUNTIME:
+    settings.DEFAULT_LIBRARY_FUNCS_TO_INCLUDE.append('$addOnPreRun')
+  if 'postRun' in settings.INCOMING_MODULE_JS_API and not settings.MINIMAL_RUNTIME:
+    settings.DEFAULT_LIBRARY_FUNCS_TO_INCLUDE.append('$addOnPostRun')
```

Comment:
How about putting these 4 lines in a `if not MINIUMAL_RUNTIME` block?

Reply:
Are there not other INCOMING_MODULE_JS_API elements that need to be included here?

Reply:
PreRun and PostRun are the only ones that can come from the Module object in this style:
```js
#if expectToReceiveOnModule('preRun')
  if (Module['preRun']) {
    if (typeof Module['preRun'] == 'function') Module['preRun'] = [Module['preRun']];
    while (Module['preRun'].length) {
      addOnPreRun(Module['preRun'].shift());
    }
  }
#endif
```

Reply:
It looks like at least postRun, preRun and preInit are in the INCOMING_MODULE_JS_API list: https://github.com/emscripten-core/emscripten/blob/1e2d8343f4dc3273aa3788414336d0a26f66a09b/src/settings.js#L997-L1011

Maybe `preInit` should not be in that list becasuse it doesn't work?

Reply:
Actually I do see `preInit` is tests on the Module API.

Reply:
See test_preinit in test_other.py

Reply:
Oh, but it doesn't use callruntimecallbacks

Reply:
Ah yes.. I wonder why?  I think I tried to make it use `callRuntimeCallbacks` once and failed.. I can't remember why that didn't work.

Presumably it should?

Reply:
I see.  Fair enough maybe lets just leave that as then.

Reply:
On another note, this does make we wonder if we really need preInit and preRun. They look like they will run basically one after the other.